modules/php/Cards/Actions/ActionLetsDoThatAgain.php

```diff
@@ -20,64 +20,83 @@ public function __construct($cardId, $uniqueId)
 
   public function immediateEffectOnPlay($player_id)
   {
-    return parent::immediateEffectOnPlay($player_id);
-    // nothing now, needs to go to resolve action state
-    // @TODO: send notification to open up the discard pile and make it visible?
-  }
+    $game = Utils::getGame();
 
-  public function resolvedBy($player_id, $args)
-  {
-    $option = $args["option"];
-    $cardIdsSelected = $args["cardIdsSelected"];
+    $rulesInDiscard = $game->cards->getCardsOfTypeInLocation(
+      "rule",
+      null,
+      "discard"
+    );
 
-    // verify args has 1 card id, and it is an Action or Rule card in the discard
-    // (or that none are available in the discard pile)
-    $game = Utils::getGame();
     $actionsInDiscard = $game->cards->getCardsOfTypeInLocation(
       "action",
       null,
       "discard"
     );
+
+    if (count($rulesInDiscard) == 0 && count($actionsInDiscard) == 0) {
+      // no rules or actions in the discard, this action does nothing
+      $game->notifyAllPlayers(
+        "",
+        clienttranslate(
+          "There are no rule or action cards in the discard pile!"
+        )
+      );
+
+      return;
+    }
+
+    return parent::immediateEffectOnPlay($player_id);
+  }
+
+  public function resolveArgs()
+  {
+    $game = Utils::getGame();
+
     $rulesInDiscard = $game->cards->getCardsOfTypeInLocation(
       "rule",
       null,
       "discard"
     );
-    if ($actionsInDiscard == 0 && $rulesInDiscard == 0) {
-      // nothing available in discard, this action does nothing
-      return;
-    }
 
-    if (count($cardIdsSelected) != 1) {
-      Utils::throwInvalidUserAction(
-        fluxx::totranslate(
-          "You must select exactly 1 Action or New Rule card from discard pile"
-        )
-      );
-    }
+    $actionsInDiscard = $game->cards->getCardsOfTypeInLocation(
+      "action",
+      null,
+      "discard"
+    );
+
+    return [
+      "discard" => $rulesInDiscard + $actionsInDiscard,
+    ];
+  }
+
+  public function resolvedBy($player_id, $args)
+  {
+    $game = Utils::getGame();
+
+    $card = $args["card"];
+    $card_definition = $game->getCardDefinitionFor($card);
+
+    $cardType = $card["type"];
 
-    $cardId = $cardIdsSelected[0];
-    $cardSelected = $game->cards->getCard($cardId);
-    $cardType = $cardSelected["type"];
     if (
-      $cardSelected == null ||
-      $cardSelected["location"] != "discard" ||
-      ($cardType != "action" && $cardType != "rule")
+      $card["location"] != "discard" ||
+      !in_array($cardType, ["rule", "action"])
     ) {
       Utils::throwInvalidUserAction(
-        fluxx::totranslate(
-          "You must select exactly 1 Action or New Rule card from discard pile"
+        clienttranslate(
```

Comment:
Are you sure this clienttranslate does the same?
From the info at the end of https://en.doc.boardgamearena.com/Translations I gathered that for exceptions we do need to use the self::_() instead of the clienttranslate.
That's why for our modules I added that public totranslate wrapper for the protected _()

Reply:
Actually you might be right, I'm not sure why I changed this. I'll fix it later today.